url/url.h

```diff
@@ -51,6 +53,12 @@ struct Url {
     std::variant<std::string, std::vector<std::string>> path;
     std::optional<std::string> query;
     std::optional<std::string> fragment;
+
+    std::string serialize(bool exclude_fragment = false) const;
+    std::string serialize_path() const;
+
+    // https://url.spec.whatwg.org/#url-equivalence
+    inline bool operator==(Url const &b) { return serialize() == b.serialize(); }
```

Comment:
No need to mark member functions with `inline` when defined where they're declared like this.

Reply:
The function should be marked const, or it will requires a mutable `Url` to be comparable.